### DISCUSSION TITLE
Does milvus compiled from the source support the use of Gpus or domestic processors?

### DISCUSSION THREAD

**Discussion by break0123** [milvus-io/milvus#36512] [Q&A and General discussion]
I'm running milvus standalone on a computing platform in China, so I can't use nvidia tools.I tried to compile from source files, encountered a lot of compilation errors, and I didn't know how to use the gpu after compiling.Source code compilation related to the relevant information can be found is limited, please ask you can further supplement.
Also, I'm confused about how the gpu is used.If I install a standalone version of milvus directly, but map the system package of the gpu and set the environment variables, can I use the gpu?

**Comment by yhmo** [comment]
Believe me, compiling milvus with gpu is not easy work. Both milvus source code and the engine "[knowhere](https://github.com/zilliztech/knowhere)" need to be built with GPU. It is recommended to install with release gpu versions.
The instalation and configuration guide: https://milvus.io/docs/install_standalone-docker-compose-gpu.md

**Reply by break0123** [reply]
Yes, it is difficult, but I still want to ask about the feasibility of source code compilation.I saw [your post](https://cloud.tencent.com/developer/article/2326764) about compiling version 2.3.0 using make milvus-gpu.At present, does the compilation have to rely on cuda toolkit to carry out? What specific support is needed?And what's the difference between milvus in the gpu image and the source package?

**Reply by yhmo** [reply]
So far as I know, cuda toolkit is required. The article is about v2.3, you can compile v2.3 source by following the article. The project is always changed, v2.4 requirement and dependecies might different.
Both GPU version and CPU version share the same repo, only the compile options are different.
If you check the scripts of the repos, you can see the script for gpu:
https://github.com/zilliztech/knowhere/tree/main/scripts
https://github.com/milvus-io/milvus/blob/954d8a5f686280dbe17d91827cd3b0a236580309/scripts/core_build.sh#L96

But I am not a expert on this area, so I could not help much.

**Reply by break0123** [reply]
Thanks for your response, this does seem difficult to use in a non-CUDA environment.Looking forward to more options for open source.
